src/payloads/AppUser.ts

```diff
@@ -1,20 +1,35 @@
-import { IsInt, IsString, IsEmail } from 'class-validator';
+import { IsEnum, IsInt, IsString, IsEmail } from 'class-validator';
+import { AppUserRole } from '@Entities';
 
 export class AppUserPKPayload {
   @IsInt()
   readonly id: number;
 }
 
-export class BaseAppUserPayload {
+abstract class BaseAppUserPayload {
   @IsString()
   readonly firstName: string;
 
   @IsString()
   readonly lastName: string;
 
+  @IsString()
+  readonly major: string;
+}
+
+// If this class is empty, then it won't be decorated by class-validator, which will break docs
```

Comment:
for sth like this you could also have Response directly extend Request - no need to change just a suggestion

Reply:
honestly though - cleanest way might just be to screw all inheritance and duplicate fields between req/res

Reply:
I agree, it will be longer, but probably more clarity and we won't have to deal with the problems I had